wishlistApp/views.py

```diff
@@ -90,15 +90,21 @@ def new_item(request):
             #(not sure if saving the form OR putting info into an Item obj is correct)
             '''
             name = request.POST['itemN']
-            url = request.POST['itemURL']
+            itemURL = request.POST['itemURL']
             desc = request.POST['itemD']
             image = request.POST['imageURL']
             priority = request.POST['itemP']
-            ins = Item(name=name, image=url, description=desc, priority = priority, user_id=request.user.get_username())
+            for urls in URL.objects.all():
+                if urls == itemURL:
+                    ins = Item(name=name, image=url, description=desc, priority = priority, user_id=request.user.get_username(), url_id=urls)
+                    break
+                else:
+                    ins = Item(name=name, image=url, description=desc, priority = priority, user_id=request.user.get_username(), url_id=itemURL)
+                    break
             ins.save()
             '''
             messages.success(request, f"Item has been added to your Wishlist!")
-            context = {
-                'form' : form,
-            }
-            return render(request, 'wishlistApp/newItem.html', context)
+            return render(request, 'wishlistApp/newItem.html', {'form': form})
+        else:
+            form = ItemUpdateForm(request.POST, instance=request.user)
```

Comment:
@licup 
Since the route doesn't actually show anything, I don't have anything I can test so this is just from my understanding of the code, but from what I see, I think the two returns might be switched? The logic right now is saying if form is invalid, `form = ItemUpdateForm`? which I believe is pushing the updated form to the database? Shouldn't we only want to do that in the if `form.is_valid()` part?

Reply:
oh for that i must've messed up on my formatting, i meant to put that in the same tab space as the ### if request.method == 'POST': ###